app/controllers/oauth_controller.rb

```diff
@@ -69,4 +69,32 @@ def destroy_omniauth
 
     redirect_to root_url
   end
+
+  private
+
+    def is_same_currency
+      connec_client = Maestrano::Connector::Rails::ConnecHelper.get_client(current_organization)
+      connec_response_hash = JSON.parse(connec_client.get('company').body)
+      @connec_currency = connec_response_hash.dig('company', 'currency')
+
+      shopify_client = Maestrano::Connector::Rails::External.get_client(current_organization)
+      shopify_response_hash = shopify_client.find('Shop').first
+      @shopify_currency = shopify_response_hash['currency']
+
+      return @connec_currency.blank? || @shopify_currency.blank? || @shopify_currency == @connec_currency
+    rescue => e
+      Maestrano::Connector::Rails::ConnectorLogger.log('warn', current_organization, "Error when comparing currencies: #{e}")
+      true
+    end
+
+    def render_create_omniauth
+      return render html: "<script>alert('#{alert_msg}'); window.location.assign('/home/index');</script>".html_safe unless is_same_currency
```

Comment:
It would be nicer if you used the `flash` scope http://guides.rubyonrails.org/action_controller_overview.html#the-flash
```ruby
def render_create_omniauth
  flash[:notice] = 'blabla' unless is_same_currency
  redirect_to root_url
end
```